MapleServer2/Types/QuestStatus.cs

```diff
@@ -1,33 +1,72 @@
 ﻿using System.Collections.Generic;
 using Maple2Storage.Types.Metadata;
+using MapleServer2.Data.Static;
+using MapleServer2.Database;
 
 namespace MapleServer2.Types
 {
     public class QuestStatus
     {
-        public readonly long Uid;
-        public Player Player;
-        public QuestBasic Basic { get; set; }
+        public long Uid { get; private set; }
+        public int Id { get; private set; }
+        public QuestBasic Basic { get; private set; }
         public bool Started { get; set; }
         public bool Completed { get; set; }
         public long StartTimestamp { get; set; }
         public long CompleteTimestamp { get; set; }
-        public long StartNpcId { get; set; }
-        public long CompleteNpcId { get; set; }
-        public List<QuestCondition> Condition { get; set; }
-        public QuestReward Reward { get; set; }
-        public List<QuestRewardItem> RewardItems { get; set; }
+        public long StartNpcId { get; private set; }
+        public long CompleteNpcId { get; private set; }
+        public List<Condition> Condition { get; set; }
+        public QuestReward Reward { get; private set; }
+        public List<QuestRewardItem> RewardItems { get; private set; }
+
+        public readonly Player Player;
 
         public QuestStatus() { }
 
-        public QuestStatus(QuestMetadata metadata)
+        public QuestStatus(Player player, QuestMetadata metadata)
         {
+            Player = player;
+            Id = metadata.Basic.Id;
             Basic = metadata.Basic;
             StartNpcId = metadata.StartNpc;
             CompleteNpcId = metadata.CompleteNpc;
-            Condition = metadata.Condition;
+            Condition = new List<Condition>();
+            foreach (QuestCondition condition in metadata.Condition)
+            {
+                Condition.Add(new Condition(condition.Type, condition.Codes, condition.Goal, 0, condition.Target));
+            }
             Reward = metadata.Reward;
             RewardItems = metadata.RewardItem;
+            Uid = DatabaseManager.AddQuest(this);
+        }
+
+        public void SetMetadataValues(int id)
+        {
+            QuestMetadata metadata = QuestMetadataStorage.GetMetadata(id);
+            Basic = metadata.Basic;
+            StartNpcId = metadata.StartNpc;
+            CompleteNpcId = metadata.CompleteNpc;
+            Reward = metadata.Reward;
+            RewardItems = metadata.RewardItem;
+        }
+    }
+
+    public class Condition
```

Comment:
New class?

Reply:
Didn't want to change values inside the metadata class. If you feel it's ok I can revert it to keep using the metadata class

Reply:
Oh gotcha, nevermind